CONTRIBUTING.md

```diff
@@ -28,22 +28,27 @@ to search for existing bugs before you create another one. Contributions are alw
 ## Submitting a contribution
 You can propose contributions by sending pull requests through GitHub. Following these guidelines
 will help us to merge your pull requests smoothly:
+
 1.  It is generally a good idea to file an issue to explain your idea before writing
 code or submitting a PR, especially when introducing new features. Trivial (must be
 single commit and very obvious change) PRs may be accepted without an issue if the
 change really is simple, but committers have the right to request an issue for any
-change.
+change. If in doubt, create an issue.
 2. Please read carefully and adhere to the legal considerations and copyright/license requirements outlined below.
 3. Follow the coding style and format of the code you are modifying (see doc/CodingStandard.md)
-4. All commits must be signed (see Legal considerations, below) and use the following commit template (Issue #nn may be omitted if appropriate):
+4. All commits must be signed (see Legal considerations, below) and use the following commit template ("#n" only needed if there is an issue):
+
     ```
-    Issue #nn Simple Description
+    Short, simple summary that must include #n
+
     Detailed information if required
 
     Signed-off-by: First Last <email>
     ```
+
 5. Ensure that "make test" passes all tests before you submit a Pull Request.
-6. Reference the Issue number when you create the Pull Request comment.
+6. The pull request comment should reference the issue number in its title, just as in the commit message template.
```

Comment:
For the github cross-referencing to work, the issue number has to be in a discussion comment of the pull request, and not just the pull request title.  Almost everyone already puts the issue number in the pull request title now, and we get no cross-ref.